### ISSUE TITLE
Input/Output device label "Name" shouldn't be wrapped character by character

### ISSUE THREAD

**Issue by bdaase** [wwmm/easyeffects#1372]
### EasyEffects Version

6.2.2

### What package are you using?

Flatpak (Flathub)

### Distribution

Fedora 35

### Describe the bug

The input and output device label "Name" shouldn't be wrapped character by character.
It currently looks as follows:
![image](https://user-images.githubusercontent.com/24682367/151651808-9f701b05-c30b-496f-b0e7-bba19661dad6.png)

This might be caused by the relatively long node name.

### Debug Log

<details><summary>Debug Log</summary><br/>

```
Paste your log here
```
</details>

**Comment by vchernin** [comment]
> This might be caused by the relatively long node name.

I think your theory is correct:

![image](https://user-images.githubusercontent.com/38842733/151652122-fcad96ab-d435-4017-beea-afad3d277f3c.png)

The node description here happens to be slightly shorter.

**Comment by wwmm** [comment]
The `Name` label is set as the `title` property of a libadwaita `AdwActionRow` https://github.com/wwmm/easyeffects/blob/9fd19c8195905be17e9d10f71565f418648c68a6/data/ui/pipe_manager_box.ui#L52. We could try to force some kind of workaround but I think it may be better to take this to libadwaita developers so a solution that benefits all gtk applications can be found. We are not doing anything different from libadwaita examples. It is just a `AdwActionRow` with a title and a `GtkDropDown` inside of it.

**Comment by bdaase** [comment]
@wwmm Alright, thanks for the info.

Would you please open the libadwaita issue as you know all the technical deatils and therefore could create a way more meaningful issue?

**Comment by wwmm** [comment]
>Would you please open the libadwaita issue as you know all the technical deatils and therefore could create a way more meaningful issue?

I can do that.

**Comment by wwmm** [comment]
Link to the issue https://gitlab.gnome.org/GNOME/libadwaita/-/issues/409

**Comment by Digitalone1** [comment]
I like more the actual DropDown rather than the one suggested in the libadwaita issue. The change could also break uniformity because we are using the same DropDown in plugin interfaces too. 

I would suggest to put everything in one row only. The DropDown on the left and a ToggleButton `Use Default` on the right (we also use ToggleButton in plugin interfaces to enable device DropDown). Only the DropDown should be wrapped... or at lest a test should be made to verify that.

**Comment by wwmm** [comment]
> The change could also break uniformity because we are using the same DropDown in plugin interfaces too.

Considering the issue I found  in the AdwComboRow I would not use them anyway. But I do no think we have to use the same kind of dropdown everywhere. 

>I would suggest to put everything in one row only.

I do not see how that helps. The source of our problem is exactly not having enough space on the horizontal. And I think that the switch is fine in it's own row.

**Comment by wwmm** [comment]
Looking at the documentation of the `AdwActionRow` objects we are already using I saw that they accept an icon in the place of the title string. So one option is changing the interface to something like this
![Screenshot from 2022-01-30 10-56-37](https://user-images.githubusercontent.com/16225172/151702732-09ad167e-c1f8-4360-a02e-2e5f06c2cca7.png)
At least to me it looks fine and we won't have to worry about the title string being broken in several lines.

**Comment by Digitalone1** [comment]
> I do not see how that helps. The source of our problem is exactly not having enough space on the horizontal. And I think that the switch is fine in it's own row.

We reduce vertical height and the ToggleButton label would not be wrapped. There will be always a device description too long for the DropDown, but at least the label inside the ToggleButton is intact.

**Comment by wwmm** [comment]
>We reduce vertical height and the ToggleButton label would not be wrapped. There will be always a device description too long for the DropDown, but at least the label inside the ToggleButton is intact.

I see. It would work.  But now that I saw that there is a `icon-name` property besides the `title` property I think it is better to have the icon that was inside the dropdown in the `AdwActionRow`

**Comment by wwmm** [comment]
I have updated the master branch replacing the title property by an icon-name. It is the simplest thing to do.

**Comment by bdaase** [comment]
Alright, thanks. Looks like that quite cleverly workarounds the problem 👍 

So should we close this issue then?

**Comment by wwmm** [comment]
>So should we close this issue then?

I think we can.

**Comment by Digitalone1** [comment]
It's good for device descriptions with long names, but not so good for short ones:

![Schermata del 2022-01-30 20-30-43](https://user-images.githubusercontent.com/25790525/151714565-4f0023c8-ea61-4c37-a531-fd40923fd984.png)

Too much space wasted, I keep preferring the solution of one row with ToggleButton.

Anyway with actual two rows we could at least [expand](https://docs.gtk.org/gtk4/property.Widget.hexpand.html) the DropDown to fit the box width, this way we always have the icon near the device name similar the previous version.

**Comment by wwmm** [comment]
>I keep preferring the solution of one row with ToggleButton.

I think it was like this in the past. But I never really liked to have both in the same row.

>Anyway with actual two rows we could at least expand the DropDown to fit the box width

I will try to see if libadwaita will not get in the way. Depending on the situation one approach would be to have the icon inside the dropdown but not having a title in the `AdwActionRow`. I don't think we need one there. THe dropdow with the is already enough to tell the user what they do.

**Comment by wwmm** [comment]
>I will try to see if libadwaita will not get in the way

It gets in the way because it reserver a space for the row title even if I do not set one. But totally by accident I have found a way to avoid that line break in the row title hahaha XD
![Screenshot from 2022-01-30 17-05-28](https://user-images.githubusercontent.com/16225172/151715714-50ca6c08-00b2-4f5b-a070-e353caafa80e.png)
The `AdwActionRow` has a property named `title-lines`. In the image above I set it in the output device dropdown.
